### ISSUE TITLE
[DatePicker] allowFreeInput: does not work with custom format

### ISSUE THREAD

**Issue by devj3ns** [mantinedev/mantine#3405]
### What package has an issue

@mantine/dates

### Describe the bug

When using the German date format (`DD.MM.YYYY`) in the `DatePicker` component the first digit (in germany the day) is used as the month, which means dates such as 16.11.2001 cannot be entered because there is no 16th month.

```html
<DatePicker
      locale="de"
      inputFormat="DD.MM.YYYY"
      allowFreeInput
    />
```

https://user-images.githubusercontent.com/20878653/213874021-b339d602-9254-4201-8029-c9860112c841.mp4

As you can see in the video, it is not possible to enter the date 16.11.2001 or 31.01.2023 which are valid dates in the german date format.

### What version of @mantine/hooks page do you have in package.json?

5.1.4

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/mantine-bug-report-gc39hy?file=/src/App.tsx

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

**Comment by rtivital** [comment]
The issue will be fixed in 6.0 release.

**Comment by rtivital** [comment]
Fixed in 6.0: DateInput component

**Comment by KilianB** [comment]
It should be noted that the when using the value format we had to extends dayJs with the customParseFormat in order for this to work. This should be mentioned in the docs

````ts
import customParseFormat from 'dayjs/plugin/customParseFormat';
dayjs.extend(customParseFormat)
````
